atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/AdjustableSweepBatchConfigSource.java

```diff
@@ -85,4 +86,18 @@ public void decreaseMultiplier() {
                 SafeArg.of("maxCellTsPairsToExamine", lastBatchConfig.maxCellTsPairsToExamine()),
                 SafeArg.of("batchSizeMultiplier", batchSizeMultiplier));
     }
+
+    private void reduceBatchSizeMultipler() {
+        SweepBatchConfig config = getRawSweepConfig();
+        double smallestSensibleBatchSizeMultiplier =
+                1.0 / NumberUtils.max(
+                        config.maxCellTsPairsToExamine(), config.candidateBatchSize(), config.deleteBatchSize());
+
+        double newBatchSizeMultiplier = batchSizeMultiplier / 2;
+        if (newBatchSizeMultiplier < smallestSensibleBatchSizeMultiplier) {
+            batchSizeMultiplier = smallestSensibleBatchSizeMultiplier;
```

Comment:
Agree that logging here would be useful. I have concerns that this might log _too much_ like the clock skew monitor though.

Out of interest, do we know offhand what happens if C* is unreachable or down? For a timeout exception you get automatic log throttling, but uncertain about connect exceptions that arise more quickly.

Reply:
Done, I've also added a guard so if the `batchSizeMultiplier` is already set to the `smallestSensibleValue` then we early out to avoid excessive logging.